strimzi-operator/src/test/java/io/spoud/kcc/operator/OperatorTest.java

```diff
@@ -0,0 +1,283 @@
+                              String resourceName, List<AclOperation> allowedOperations, List<AclOperation> deniedOperations) {
+        return new KafkaUserBuilder()
+                .withNewMetadata()
+                .withName(username)
+                .withAnnotations(annotations)
+                .endMetadata()
+                .withNewSpec()
+                .withAuthorization(
+                        new KafkaUserAuthorizationSimpleBuilder()
+                                .addNewAcl()
+                                .withOperations(allowedOperations)
+                                .withType(AclRuleType.ALLOW)
+                                .withResource(
+                                        new AclRuleTopicResourceBuilder()
+                                                .withPatternType(patternType)
+                                                .withName(resourceName)
+                                                .build()
+                                )
+                                .endAcl()
+                                .addNewAcl()
+                                .withOperations(deniedOperations)
+                                .withType(AclRuleType.DENY)
+                                .withResource(
+                                        new AclRuleTopicResourceBuilder()
+                                                .withPatternType(patternType)
+                                                .withName(resourceName)
+                                                .build()
+                                )
+                                .endAcl()
+                                .build()
+                )
+                .endSpec()
+                .build();
+    }
+
+    void addKafkaTopic(KafkaTopic t) {
+        client.resources(KafkaTopic.class)
+                .inNamespace(config.namespace())
+                .resource(t)
+                .createOr(NonDeletingOperation::update);
+    }
+
+    KafkaTopic getTopicInstance(String topicName, String appName) {
+        return new KafkaTopicBuilder()
+                .withNewMetadata()
+                .withName(topicName)
+                .withAnnotations(Map.of(
+                        config.contextAnnotationPrefix() + TOPIC_APP_KEY, appName
+                ))
+                .endMetadata()
+                .withNewSpec()
+                .withPartitions(1)
+                .withReplicas(1)
+                .endSpec()
+                .build();
+    }
+
+    public interface ThrowableRunnable {
+        void run() throws Exception;
+    }
+
+    private void delayedAsyncRun(ThrowableRunnable r) {
+        CompletableFuture.runAsync(() -> {
+            try {
+                Thread.sleep(1000); // give downstream code some time to prepare
+                r.run();
+            } catch (Exception e) {
+                e.printStackTrace();
+                fail(e);
```

Comment:
nitpick I don't think you need to print the stack if you fail just after

```suggestion
                fail(e);
```